plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -60,73 +63,82 @@
 import java.io.IOException;
 import java.util.*;
 
-public class PipelineStore extends AbstractLifecycleComponent {
+public class PipelineStore extends AbstractComponent implements Closeable {
 
     public final static String INDEX = ".ingest";
     public final static String TYPE = "pipeline";
 
-    private final ThreadPool threadPool;
     private final TimeValue scrollTimeout;
-    private final ClusterService clusterService;
     private final Provider<Client> clientProvider;
-    private final TimeValue pipelineUpdateInterval;
     private final Pipeline.Factory factory = new Pipeline.Factory();
     private final Map<String, Processor.Factory> processorFactoryRegistry;
 
     private volatile Client client;
+    private volatile boolean pipelineIndexReady = false;
     private volatile Map<String, PipelineDefinition> pipelines = new HashMap<>();
 
     @Inject
-    public PipelineStore(Settings settings, Provider<Client> clientProvider, ThreadPool threadPool, Environment environment, ClusterService clusterService, Map<String, ProcessorFactoryProvider> processorFactoryProviders) {
+    public PipelineStore(Settings settings, Provider<Client> clientProvider, Environment environment, Map<String, ProcessorFactoryProvider> processorFactoryProviders) {
```

Comment:
what is this provider doing here? We can't do this in our production code this will take hours for decouple again We either pass a client or not but not yet another indirection.